### PR TITLE
SNOW-94: Links to open up as new page

### PR DESCRIPTION
**Pull request by Psichorex**
SNOW-94

### CODE REVIEW

File: Extensions/TargetBlank/Filters/TargetBlankFilter.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            return;
+        }
+
+        // Until NodeExtensions is not being used this solution is for to replace the usage of GULP
+        // When NE will come alive this script should be refactored and store as a .js file.
+        _resourceManager.RegisterFootScript(new HtmlString(
+            "<script>function targetBlank(){const x=document.querySelectorAll('a');" +
+            "for(let i=0;i<x.length;i++){if(!x[i].href.match(/^mailto:/)&&(x[i].hostname!==location.hostname))" +
+            "{x[i].setAttribute('target','_blank')}}}window.addEventListener('load',function()" +
+            "{window.setTimeout(targetBlank,100)},false);</script>"));
```
Comment (Luko6) [inline Extensions/TargetBlank/Filters/TargetBlankFilter.cs]: You should put the string into it's own variable and use a verbatim string (`@"";`)  so you can format it in a more readable way.

File: Extensions/TargetBlank/Filters/TargetBlankFilter.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    {
+        if (context.Result is not ViewResult)
+        {
+            await next();
+
+            return;
+        }
+
+        // Until NodeExtensions is not being used this solution is for to replace the usage of GULP
+        // When NE will come alive this script should be refactored and store as a .js file.
```
Comment (Luko6) [inline Extensions/TargetBlank/Filters/TargetBlankFilter.cs]:
```suggestion
        // Until Node Extensions is ready for use, this solution is here to replace the usage of Gulp. When Node
        // Extensions is ready, this script should be replaced with a JavaScript file.
```

**Review by Luko6 — CHANGES_REQUESTED** [review]

File: Extensions/TargetBlank/Filters/TargetBlankFilter.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+        const string script = @"<script>
+                                function targetBlank() {
+                                    const x=document.querySelectorAll('a');
+                                    for(let i=0;i<x.length;i++) {
+                                        if(!x[i].href.match(/^mailto:/)&&(x[i].hostname!==location.hostname&&(!x[i].href.match('javascript:')))) {
+                                            x[i].setAttribute('target','_blank')}}}
+                                window.addEventListener('load',function() {
+                                    window.setTimeout(targetBlank,100)},false);
+                               </script>";
```
Comment (Luko6) [inline Extensions/TargetBlank/Filters/TargetBlankFilter.cs]: Let's make it more readable.

```suggestion
        const string script = @"
            <script>
                 function targetBlank() {
                     const x=document.querySelectorAll('a');
                     for(let i=0;i<x.length;i++) {
                         if(!x[i].href.match(/^mailto:/)&&(x[i].hostname!==location.hostname&&(!x[i].href.match('javascript:'))))
                         {
                             x[i].setAttribute('target','_blank');
                         }
                     }
                 }
                 window.addEventListener(
                     'load',
                     function() {
                         window.setTimeout(targetBlank,100)
                     },
                     false);
            </script>";
```

**Review by Luko6 — CHANGES_REQUESTED** [review]

File: Extensions/TargetBlank/Filters/TargetBlankFilter.cs (hunk 3 above)
Comment (Luko6) [reply Extensions/TargetBlank/Filters/TargetBlankFilter.cs]: You can simply press `Commit suggestion`, that way you won't make mistakes while copy and pasting for example here the lines between the script tags being idented by one more tab than in my suggestion. Please fix.

**Review by Luko6 — CHANGES_REQUESTED** [review]
Fix identation between script tags

File: Extensions/TargetBlank/Filters/TargetBlankFilter.cs (hunk 3 above)
Comment (0liver) [reply Extensions/TargetBlank/Filters/TargetBlankFilter.cs]: [Added benefit](https://lombiq.atlassian.net/wiki/spaces/DEV/pages/1042939995/Doing+code+reviews#:~:text=an%20added%20benefit%20is%20that%20such%20suggestions%20can%20be%20applied%20with%20one%20click%20by%20the%20author%2C%20automatically%20resolving%20the%20comment%20too%2C%20so%20you%20don%E2%80%99t%20have%20to%20check%20it%20again.).

File: Extensions/TargetBlank/Filters/TargetBlankFilter.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+                            }
+                        }
+                    }
+                    window.addEventListener(
+                        'load',
+                        function() {
+                            window.setTimeout(targetBlank,100)
+                        },
+                        false);
+            </script>";
```
Comment (Luko6) [inline Extensions/TargetBlank/Filters/TargetBlankFilter.cs]: Let's make it more readable.

```suggestion
        const string script = @"
            <script>
                 function targetBlank() {
                     const x=document.querySelectorAll('a');
                     for(let i=0;i<x.length;i++) {
                         if(!x[i].href.match(/^mailto:/)&&(x[i].hostname!==location.hostname&&(!x[i].href.match('javascript:'))))
                         {
                             x[i].setAttribute('target','_blank');
                         }
                     }
                 }
                 window.addEventListener(
                     'load',
                     function() {
                         window.setTimeout(targetBlank,100)
                     },
                     false);
            </script>";
```

**Review by Luko6 — CHANGES_REQUESTED** [review]

File: Extensions/TargetBlank/Filters/TargetBlankFilter.cs (hunk 3 above)
Comment (0liver) [reply Extensions/TargetBlank/Filters/TargetBlankFilter.cs]: Also, FYI, using [`bind`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind), this:

```js
function() {
    window.setTimeout(targetBlank,100)
}
```

can be replaced by the following statement, which achieves exactly the same:

```js
window.setTimeout.bind(window, targetBlank, 100)
```

Comment (0liver) [reply Extensions/TargetBlank/Filters/TargetBlankFilter.cs]: Please also add spaces after `,` and `;`, around `=` etc.

Best would probably be to use a regular `.js` file, embed it into the module, and load it as an embedded resource. Worse, but still better than JS inside of C#, you could put into a View and render it as a string here using Orchard foo (@Luko6 probably knows how; we did the same in O1).

Comment (Luko6) [reply Extensions/TargetBlank/Filters/TargetBlankFilter.cs]: There is a comment already explaining why this isn't in a seperate `.js` file and when will it be in one, no need to waste time and resources on it now.

**Review by Luko6 — APPROVED** [review]

**Pull request merged**